docs/src/references/tedge-configuration-management.md

```diff
@@ -0,0 +1,365 @@
+* In summary, the responsibilities of the plugin are:
+  * to define the list of files under configuration management
+  * to notify the local MQTT bus when this list is updated,
+  * to upload these files to the tedge file transfer repository on demand,  
+  * to download the files pushed from the tedge file transfer repository,
+  * to make sure that the target files are updated atomically after successful download,
+  * to notify the device software when the configuration is updated.
+* By contrast, the plugin is not responsible for:
+  * checking the uploaded files are well-formed,
+  * restarting the configured processes,
+  * establishing any direct connection to clouds.
+* A user-specific component, installed on the device,
+  can implement more sophisticated configuration use-cases by:
+  * listening for configuration updates on the local thin-edge MQTT bus,
+  * restarting the appropriate processes when appropriate,  
+  * declaring intermediate files as the managed files,
+    to have the opportunity to check or update their content
+    before moving them to the actual targets.
+
+## Installation
+
+As part of this plugin installation:
+* On systemd-enabled devices, the service definition file for this `tedge-configuration-plugin` daemon is also installed.
+
+Once installed, the `tedge-configuration-plugin` runs as a daemon on the device,
+listening to configuration snapshot commands on the `<root>/<identifier>/cmd/config_snapshot/+` [MQTT topic](mqtt-api.md#command-examples) 
+and to configuration update commands on the `<root>/<identifier>/cmd/config_update/+` [MQTT topic](mqtt-api.md#command-examples).
+
+## Configuration
+
+The `c8y-configuration-plugin` configuration is stored by default under `/etc/tedge/plugins/tedge-configuration-plugin.toml`
+
+This [TOML](https://toml.io/en/) file defines the list of files to be managed by the plugin.
+Each configuration file is defined by a record with:
+* The full `path` to the file.
+* An optional configuration `type`. If not provided, the `path` is used as `type`.
+  This `type` is used to declare the supported configuration file and then to trigger operations on that file.
+  All the configuration `type`s are declared to the local MQTT bus on start
+  and on change of the `plugins/tedge-configuration-plugin.toml` file.
+* Optional unix file ownership: `user`, `group` and octal `mode`.
+  These are only used when a configuration file pushed from a config_update command doesn't exist on the device.
+  When a configuration file is already present on the device, this plugin never changes file ownership,
+  ignoring these parameters.
+
+```sh
+cat /etc/tedge/plugins/tedge-configuration-plugin.toml
+```
+
+```toml title="file: /etc/tedge/plugins/tedge-configuration-plugin.toml"
+files = [
+  { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },
+  { path = '/etc/tedge/mosquitto-conf/c8y-bridge.conf' },
+  { path = '/etc/tedge/mosquitto-conf/tedge-mosquitto.conf' },
```

Comment:
I would prefer to always specify a `type` field in the examples, as using the `path` as the `type` is a very bad user experience as it enforces the user to have to know where the file is being applied to the device instead of what the configuration type represents.

The user does not care where the file is stored, they just care about applying configuration, and the plugin already knows where to write the file. It also allows for different devices to define different paths for the same type in case the path changes depending on the device type (across different operating systems, the path might change).